### DISCUSSION TITLE
[auth0-client][authorization-service] machine 2 machine authentication integration with auth0

### DISCUSSION THREAD

**Discussion by Epado** [mia-platform/community#219] [Q&A] [answered]
Hi all, i would like to allow users and servers to authenticate to the same apis. 
At the moment my application exposes apis to the public which are closed and need authentication, such authentication is done using auth0. I would like to add a feature to that, and allow servers or businesses to authenticate to the same apis, using the same authentication provider. In order to do it i had a look at the machine 2 machine authentication flow in auth0 (https://auth0.com/blog/using-m2m-authorization/), and i would say that the access token generation part is pretty straight-forward. 
What i am actually missing is how do i integrate it with the authorization -> auth0-cient flow, hence allowing a server to make a call to one of my apis but with the bearer token provided by the direct call to auth0. 
Thanks in advance,

**Comment by davidebianchi** [comment, marked as answer]
Hi @Epado. Using our authorization-service, you should give an endpoint which verify the authorization.
So, probably for this use case you should create a custom service which verify the auth. From the marketplace, is also present a client-credentials service which could handle the m2m authentication integrated with authorization-service if you want to take a look.
